source/h1_connection.c

```diff
@@ -615,6 +627,50 @@ static void s_stream_complete(struct aws_h1_stream *stream, int error_code) {
     aws_http_stream_release(&stream->base);
 }
 
+static void s_add_time_measurement_to_stats(uint64_t start_ns, uint64_t end_ns, uint64_t *output_ms) {
+    if (end_ns > start_ns) {
+        *output_ms += aws_timestamp_convert(end_ns - start_ns, AWS_TIMESTAMP_NANOS, AWS_TIMESTAMP_MILLIS, NULL);
+    }
+}
+
+static void s_set_outgoing_stream_ptr(struct h1_connection *connection, struct aws_h1_stream *next_outgoing_stream) {
+    struct aws_h1_stream *prev = connection->thread_data.outgoing_stream;
```

Comment:
Just a heads-up that when we change the outgoing_stream_ptr, it doesn't necessarily mean that the stream's data is done writing to the socket. It just means we've copied the stream's last outgoing bytes to a `aws_channel_message`. So for a stream whose outgoing data all fits in 16KB, it's only going to be the "current outgoing stream" for a fleeting moment, even if the socket ends up taking minutes to actually write that full `aws_channel_message`

Maybe what we want to measure is the time from when we shipped the `aws_channel_message` down the channel, to when it finished writing (there is a callback when the write completes, but the H1 code isn't currently hooked into it)

Reply:
That would be a precision improvement, but how complicated would its addition be?  Seems like something that could wait.